src/Protocol/Http/Authenticator/JwtAuthenticator.php

```diff
@@ -30,24 +30,50 @@ class JwtAuthenticator implements AuthenticatorInterface
      */
     private $jwt;
 
+    /**
+     * @var string
+     */
+    private $jws;
+
+    /**
+     * @var \DateInterval
+     */
+    private $jwsLifetime;
+
+    /**
+     * @var \DateTime
+     */
+    private $jwsValidTo;
+
     /**
      * Constructor.
      *
-     * @param JwtInterface $jwt
+     * @param JwtInterface  $jwt
+     * @param \DateInterval $jwsLifetime
+     * @throws \InvalidArgumentException
      */
-    public function __construct(JwtInterface $jwt)
+    public function __construct(JwtInterface $jwt, \DateInterval $jwsLifetime = null)
     {
         $this->jwt = $jwt;
+        if ($jwsLifetime && $jwsLifetime->invert) {
+            throw new \InvalidArgumentException('JWS lifetime must not be inverted');
+        }
+
+        $this->jwsLifetime = $jwsLifetime;
     }
 
     /**
      * {@inheritdoc}
      */
     public function authenticate(Request $request): Request
     {
-        $jws = $this->createJwsContent();
+        $now = new \DateTime();
```

Comment:
Maybe this code is hard for understanding.

Maybe better:

```php
$now = new \DateTimeImmutable();

if (!$this->jws || $this->jwsValidTo < $now) {
    // Regenerate or create new JWS token.

    $this->jwsValidTo = $this->jwsLifetime ? $now->add($this->jwsLifetime) : null;
}
```

1. We use only `jws` and `valid to` in if condition. The `valid to` will be null if the lifetime is empty.
2. Set the null to `valid to` if the lifetime is empty.
3. Use `DateTimeImmutable` (this better for work with objects for me).

I think, what this code easy for reading and understanding.